test/framework-ondata-mocha.js

```diff
@@ -950,5 +950,119 @@ should.be.rejectedWith('No wrapper fRunUpdater function found!  Cannot run updat
+                                                   isTransient: false 
+                                                 },
+                                                 {fRunUpdater: function(vni,input) { 
+                                                      nodeInstance = vni.nodeInstance;
+                                                      vni.outputState({data: testData, 
+                                                                       lm: createLm()}); 
+                                                 }}
+                )} 
+            }).then(function(network){
+
+                 var testNode = network.processes.testNode.component;
+
+                 return new Promise(function(done, fail) {
+
+		     test.onOutPortData(testNode, 'output', done);
+		     test.onOutPortData(testNode, 'error', fail);
+
+                     network.graph.addInitial("init data", 'testNode', 'input');
+                 }).then(function(done) {
+                     test.verifyState(done, '', testData);
+
+                     // Verify VNIs have been cleared
+                     expect(nodeInstance).to.not.be.empty;
+                     nodeInstance.vnis.should.be.an('object');
+                     nodeInstance.vnis.should.not.be.empty;
+                 });
+            });
+        });
+
+        it("should clear transient node VNIs after sending output state downstream", function() {
+            var nodeInstance;
+            var testData = "Ciao!";
+            return test.createNetwork({
+                testNode: {getComponent: factory({
+                                                   inPorts: {input: {}}, 
+                                                   outPorts: {output: {}},
+                                                   isTransient: true
+                                                 },
+                                                 {fRunUpdater: function(vni,input) { 
+                                                      nodeInstance = vni.nodeInstance;
+                                                      vni.outputState({data: testData, 
+                                                                       lm: createLm()}); 
+                                                 }}
+                )} 
+            }).then(function(network){
+
+                 var testNode = network.processes.testNode.component;
+
+                 return new Promise(function(done, fail) {
+
+		     test.onOutPortData(testNode, 'output', done);
+		     test.onOutPortData(testNode, 'error', fail);
+
+                     network.graph.addInitial("init data", 'testNode', 'input');
+                 }).then(function(done) {
+                     test.verifyState(done, '', testData);
+
+                     // Verify VNIs have been cleared
+                     expect(nodeInstance).to.not.be.empty;
+                     nodeInstance.vnis.should.be.an('object');
```

Comment:
This will need to be changed to: (a) use multiple VNIs; and (b) verify that the current VNI was cleared, but other VNIs were not changed.